File: backend/app/models/email_verification.py
from models.database import db
from datetime import datetime

class EmailVerify(db.Model):
    __tablename__ = "email_verification"

    id = db.Column(db.Integer, primary_key=True)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    valid_until = db.Column(db.DateTime, nullable=False)
    email_verify_code = db.Column(db.String(50), nullable=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False, unique=True)

    user = db.relationship('User', back_populates='email_verification')

File: backend/golang-app/models/email_verification.go
package models

import (
	"time"
)

type EmailVerify struct {
	ID              uint      `gorm:"primaryKey"`
	CreatedAt       time.Time `gorm:"autoCreateTime"`
	ValidUntil      time.Time `gorm:"not null"`
	EmailVerifyCode string    `gorm:"size:50;not null"`
	UserID          uint      `gorm:"not null;index"`
	User            User      `gorm:"foreignKey:UserID"`
}

func (EmailVerify) TableName() string {
	return "email_verification"
}
